sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobTestBase.java

```diff
@@ -1073,4 +1078,159 @@ protected static TokenCredential getTokenCredential(TestMode testMode) {
+                        xmlWriter.writeStringElement("Delimiter", "/");
+                    }
+                    break;
+
+                case FIND_BLOBS:
+                    elementType = "Blob";
+                    includeDummyTagsInListElement = true;
+                    startXml(xmlWriter, false);
+                    xmlWriter.writeStringElement("Where", "\"dummyKey\"='dummyValue'");
+                    xmlWriter.writeStringElement("MaxResults", String.valueOf(maxResourcesPerPage));
+                    break;
+
+                case LIST_CONTAINERS:
+                    elementType = "Container";
+                    includeDummyTagsInListElement = false;
+                    startXml(xmlWriter, false);
+                    xmlWriter.writeStringElement("MaxResults", String.valueOf(maxResourcesPerPage));
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Unknown PageType: " + pageType);
+            }
+
+            writeGenericListElement(xmlWriter, elementType, numberOfElementsOnThisPage, includeDummyTagsInListElement);
+            endXml(xmlWriter, totalPagesExpected); // This calls flush
+
+            return output.toString();
+        }
+
+        private void startXml(XmlWriter xmlWriter, Boolean hasContainerAttribute) throws XMLStreamException {
+            xmlWriter.writeStartDocument();
+            xmlWriter.writeStartElement("EnumerationResults");
+            xmlWriter.writeStringAttribute("ServiceEndpoint", "https://account.blob.core.windows.net/");
+            if (hasContainerAttribute) {
+                xmlWriter.writeStringAttribute("ContainerName", "foo");
+            }
+
+            // Write marker if not first page
+            if (pageCounter != 0) {
+                xmlWriter.writeStringElement("Marker", "MARKER--");
+            }
+        }
+
+        private void endXml(XmlWriter xmlWriter, int totalPagesExpected) throws XMLStreamException {
+            // Write NextMarker
+            if (pageCounter == totalPagesExpected - 1) {
+                xmlWriter.writeStringElement("NextMarker", null);
+            } else {
+                xmlWriter.writeStringElement("NextMarker", "MARKER--");
+            }
+
+            xmlWriter.writeEndElement(); // End EnumerationResults
+            xmlWriter.flush();
+        }
+
+        private void writeGenericListElement(XmlWriter xmlWriter, String elementType, int numberOfElementsOnThisPage,
+            Boolean includeDummyTags) throws XMLStreamException {
+            // Start elementType + s
+            xmlWriter.writeStartElement(elementType + "s");
+
+            // Write  entries
+            for (int i = 0; i < numberOfElementsOnThisPage; i++) {
+                xmlWriter.writeStartElement(elementType); // Start elementType
+                xmlWriter.writeStringElement("Name", elementType.toLowerCase() + resourceCounter++);
+
+                if (includeDummyTags) {
```

Comment:
It's a little awkward that we have a response specific parameter `includeDummyTags` in a method that is supposed to be general. Instead, we could expose a parameter in `writeGenericListElement` to accept a general `writeListElement` callable that for the `FIND_BLOBS` case we can always include this tagset writing logic?